### PR TITLE
chore: stop using piscina worker methods for runEventPipeline

### PR DESCRIPTION
**Pull request by tiina303**
## Problem

We don't really need to use piscina here and it's creating unneccesary confusion reading the code

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: plugin-server/src/worker/ingestion/event-pipeline/runner.ts [hunk 1]
```diff
@@ -94,6 +95,7 @@ export class EventPipelineRunner {
     }
 
     async runEventPipeline(event: PipelineEvent): Promise<EventPipelineResult> {
+        this.originalEvent = event
```
Comment (bretthoerner) [inline plugin-server/src/worker/ingestion/event-pipeline/runner.ts:98]: It's confusing that we store it in the constructor and also here. Does just 1 or the other not work?

**Review by bretthoerner — APPROVED** [review]
Nice, makes a lot of things easier to follow.

File: plugin-server/src/worker/ingestion/event-pipeline/runner.ts (hunk 1 above)
Comment (tiina303) [reply plugin-server/src/worker/ingestion/event-pipeline/runner.ts:98]: right, forgot to remove it from here ...  I added if before creating a function outside of the class that call locations can use

**Comment by bretthoerner** [comment]
Rebased this on master so I could try it out on my PR.

For historical records: @xvello and I noticed in https://github.com/PostHog/posthog/pull/17932 that the Piscana layer was doing funny things with our objects. What should have been an array of `Promise` objects was turned into an array of plain `Object`s... so either the promises were dropped or they were awaited, either one is scary.

We should merge https://github.com/PostHog/posthog/pull/17932 (which includes the commit from this PR) on Monday so we don't risk a behavior/performance change on Friday.

**Comment by xvello** [comment]
Thanks for the investation Brett. @tiina303 please don't merge this PR by itself, we'll merge it as part of https://github.com/PostHog/posthog/pull/17932 to avoid a regression: we'd fail to process messages that are too big for the destination topic.

**Pull request closed without merging**